src/parser/jobs/whm/modules/Lilies.tsx

```diff
@@ -0,0 +1,167 @@
+import {Plural, Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {Event} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge, TimerGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const LILY_INITIAL_STACKS = 0
+const LILY_MAX_STACKS = 3
+const LILY_60_TIME_REQUIRED = 30000
+const LILY_TIME_REQUIRED = 20000
+
+const BLOOD_LILY_BLOOM = 3
+const MISERY_COST = 3
+
+const GAUGE_FADE = 0.25
+const TIMER_FADE = 0.75
+const BLOODLILY_FADE = 0.5
+const LILY_COLOR = Color('#4f73b5')
+const BLOODLILY_COLOR = Color('#b52d6c')
+
+export class Lilies extends CoreGauge {
+
+	@dependency private suggestions!: Suggestions
+
+	private lilyConsumers = [
+		this.data.actions.AFFLATUS_RAPTURE.id,
+		this.data.actions.AFFLATUS_SOLACE.id,
+	]
+
+	private bloodLilyConsumers = [
+		this.data.actions.AFFLATUS_MISERY.id,
+	]
+
+	private lilyGauge = this.add(new CounterGauge({
+		maximum: LILY_MAX_STACKS,
+		initialValue: LILY_INITIAL_STACKS,
+		graph: {
+			label: <Trans id="whm.gauge.resource.lily">Lily</Trans>,
+			color: LILY_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+
+	private lilyTimer = this.add(new TimerGauge({
+		maximum: this.parser.patch.before('6.1') ? LILY_60_TIME_REQUIRED : LILY_TIME_REQUIRED,
+		onExpiration: this.onLilyGeneration.bind(this),
+		graph: {
+			label: <Trans id="whm.gauge.resource.lily-timer">Lily Timer</Trans>,
+			color: LILY_COLOR.fade(TIMER_FADE),
+		},
+
+	}))
+
+	private bloodLilyGauge = this.add(new CounterGauge({
+		maximum: BLOOD_LILY_BLOOM,
+		initialValue: 0,
+		graph: {
+			label: <Trans id="whm.gauge.resource.bloodlily">Blood Lily</Trans>,
+			color: BLOODLILY_COLOR.fade(BLOODLILY_FADE),
+		},
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('action').action(oneOf(this.lilyConsumers)), this.onLilySpend)
+		this.addEventHook(playerFilter.type('action').action(oneOf(this.bloodLilyConsumers)), () => this.bloodLilyGauge.spend(MISERY_COST))
```

Comment:
There's only one blood lily consumer so kinda ehhh to use matchers or `oneOf` here, but consider doing the same as I wrote above re the consts and matchers here too.